packages/go/stbernard/graph/transform.go

```diff
@@ -0,0 +1,346 @@
+			labels = append(labels, kind.String())
+		}
+
+		arrowNodes = append(arrowNodes, Node{
+			ID: node.ID.String(),
+			Position: Position{
+				X: 0,
+				Y: 0,
+			},
+			Caption:    name,
+			Label:      labels,
+			Properties: convertProperties(node.Properties.Map),
+		})
+	}
+
+	for _, edge := range edges {
+		arrowEdges = append(arrowEdges, Relationship{
+			ID:         edge.ID.String(),
+			From:       edge.StartID.String(),
+			To:         edge.EndID.String(),
+			Label:      edge.Kind.String(),
+			Properties: convertProperties(edge.Properties.Map),
+		})
+	}
+
+	return Arrows{
+		Nodes:         arrowNodes,
+		Relationships: arrowEdges,
+	}, nil
+}
+
+func convertProperties(input map[string]any) map[string]string {
+	output := make(map[string]string)
+	for key, value := range input {
+		output[key] = convertProperty(value)
+	}
+	return output
+}
+
+func convertProperty(input any) string {
+	switch v := input.(type) {
+	case string:
+		return v
+	case int:
+		return strconv.Itoa(v)
+	case float64:
+		return strconv.FormatFloat(v, 'f', -1, 64)
+	case bool:
+		return strconv.FormatBool(v)
+	case []any:
+		return strings.Join(slicesext.Map(v, convertProperty), ",")
+	case nil:
+		return "null"
+	default:
+		slog.Warn("unknown type encountered", slog.String("type", fmt.Sprintf("%T", v)))
+		return ""
+	}
+}
+
+func getNodesAndEdges(ctx context.Context, database graph.Database) ([]*graph.Node, []*graph.Relationship, error) {
+	var nodes []*graph.Node
+	var edges []*graph.Relationship
+	if err := database.ReadTransaction(ctx, func(tx graph.Transaction) error {
+		err := tx.Nodes().Filter(
+			query.Not(query.Kind(query.Node(), common.MigrationData)),
+		).Fetch(func(cursor graph.Cursor[*graph.Node]) error {
+			for node := range cursor.Chan() {
+				nodes = append(nodes, node)
+			}
+			return cursor.Error()
+		})
+		if err != nil {
+			return fmt.Errorf("error fetching nodes %w", err)
+		}
+		err = tx.Relationships().Fetch(func(cursor graph.Cursor[*graph.Relationship]) error {
+			for edge := range cursor.Chan() {
+				edges = append(edges, edge)
+			}
+			return cursor.Error()
+		})
+		if err != nil {
+			return fmt.Errorf("error fetching relationships %w", err)
+		}
+
+		return nil
+	}); err != nil {
+		return nodes, edges, fmt.Errorf("error occurred reading the database %w", err)
+	} else {
+		return nodes, edges, nil
+	}
+}
+
+func (s *command) getIngestFilePaths() ([]string, error) {
+	var (
+		testIngestFilePath = filepath.Join("cmd", "api", "src", "test", "fixtures", "fixtures", "v6", "ingest")
+		testIngestFileDir  = filepath.Join(s.root, testIngestFilePath)
+	)
+
+	if ingestFiles, err := os.ReadDir(testIngestFileDir); err != nil {
+		return []string{}, fmt.Errorf("error reading  directory %s: %w", testIngestFileDir, err)
+	} else {
+		var paths = make([]string, 0, len(ingestFiles))
+		for _, path := range ingestFiles {
+			paths = append(paths, filepath.Join(testIngestFileDir, path.Name()))
+		}
+		return paths, nil
+	}
```

Comment:
It would be more resilient if we do a filepath.WalkDir, filtering out directories and non-JSON files, and appending only json files to the list of paths.